spec/requests/physical_servers_spec.rb

```diff
@@ -10,6 +10,155 @@
         expect_single_resource_query("ems_ref" => "A59D5B36821111E1A9F5E41F13ED4F6A")
       end
     end
+
+    context "without an appropriate role" do
+      it "fails to retrieve physical servers" do
+        ps = FactoryGirl.create(:physical_server)
+
+        api_basic_authorize
+        get api_physical_server_url(nil, ps)
+
+        expect(response).to have_http_status(:forbidden)
+        expect(response.parsed_body["error"]).to include("kind" => "forbidden")
+      end
+    end
+
+    context "with valid id" do
+      it "returns both id and href" do
+        api_basic_authorize action_identifier(:physical_servers, :read, :resource_actions, :get)
+        ps = FactoryGirl.create(:physical_server)
+        ps2 = FactoryGirl.create(:physical_server)
+
+        get api_physical_server_url(nil, ps.id)
+
+        expect_single_resource_query("id" => ps.id.to_s, "href" => api_physical_server_url(nil, ps))
+      end
+    end
+
+    context "with an invalid id" do
+      it "fails to retrieve physical server" do
+        api_basic_authorize action_identifier(:physical_servers, :read, :resource_actions, :get)
+
+        get api_physical_server_url(nil, 999_999)
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context "with hardware attribute" do
+      it "retrieve hardware details" do
```

Comment:
Given that this and the following tests are largely variations on the same theme (retrieving additional attributes from the physical server), and given that request specs are in general expensive, WDYT about combining them into one call where you ask for everything at once?